src/VisualStudio/Core/Next/Remote/JsonRpcSession.cs

```diff
@@ -65,9 +65,23 @@ internal class JsonRpcSession : RemoteHostClient.Session
 
         private async Task InitializeAsync()
         {
-            // all roslyn remote service must based on ServiceHubServiceBase which implements Initialize method
-            await _snapshotClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
-            await _serviceClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
+            CancellationToken.ThrowIfCancellationRequested();
+
+            try
+            {
+                // all roslyn remote service must based on ServiceHubServiceBase which implements Initialize method
+                await _snapshotClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
```

Comment:
I don't understand.  This is our own type.  i.e. "JsonRpcClient".  We should harden the "InvokeAsync" methods of to not throw ObjectDisposedException.  You're making it so that all consumers need to know that you're leaking this out from the underlying "JsonRpc" object.

The correct location for these try/catches seems like it should be in all the  "JsonRpcClient.InvokeAsync", and not have it bleed out elsewhere.